### ISSUE TITLE
`make test` as root moving a file over `/dev/null`

### ISSUE THREAD

**Issue by lockywolf** [brucemiller/LaTeXML#2069]
How to reproduce:

```
sudo make test
```

This command is doing something that is moving a file over /dev/null, which breaks all of the system. Running tests as root is a part of some package building systems, and is unavoidable sometimes.

UPD: the actual failing test is:
```
sudo perl -Mblib t/91_latexmlc_api.t
```

UPD:`t/92_profiles.t` has the same problem
UPD :`t/93_formats.t` has the same problem
UPD: `t/95_complex_config.t` has the same problem
UPD:`t/96_fatal.t` has the same problem

I don't think this code is okay:

```
push @$opts, (['destination', "$localname.test.xml"],
    ['log',                "/dev/null"],
    ['quiet',              ''],
    ['quiet',              ''],
    ['quiet',              ''],
    ['quiet',              ''],
    ['autoflush',          1],
    ['timestamp',          '0'],
    ['nodefaultresources', ''],
    ['xsltparameter',      'LATEXML_VERSION:TEST'],
    ['nocomments',         '']);
```

I suspect that this setting doesn't actually do what you expect it to do. I suspect that it doesn't actually "pipe" output to `/dev/null`, it probably unlinks the file if it is present, and then makes a new one. It works as a user, because a user cannot delete `/dev/null`.

**Comment by dginev** [comment]
Wow, that's embarrassing. Apologies @lockywolf and thank you for catching the issue!

Looks like it's time to switch to a scratch log file in the latexml directory during testing.

**Comment by teepeemm** [comment]
Would it be better to have `lib/LaTeXML/Common/Error.pm#UseLog` check for `/dev/null`?  Or `/dev` in general?  I could see someone trying `latexml myfile --log=/dev/null`.

**Comment by tkw1536** [comment]
I think this was caused by [`bin/latexmlc`](https://github.com/brucemiller/LaTeXML/blob/bf4543cc1624d87c4a735d5560d8dadefa5c9a59/bin/latexmlc#L120), which blindly unlinks any existing file.
This code is also called by the daemon tests. 

But `/dev/null` is not a regular file, it is a device.
As such the fix is simple: Instead of checking if the file exists, check if it is a regular file. 
I've made that fix in #2094. 

Unfortunately I couldn't actually reproduce `/dev/null` being deleted - I guess my system protects against that. @lockywolf would appreciate if you could test the PR on your end and see if that indeed solves the problem.
